src/marlin_stubs/M865.cpp

```diff
@@ -36,6 +37,10 @@ void PrusaGcodeSuite::M865() {
     }
 
     FilamentType filament_type;
+    if (const auto slot = p.option<uint8_t>('H', static_cast<uint8_t>(0), static_cast<uint8_t>(adhoc_filament_type_count - 1))) {
```

Comment:
`adhoc_filament_type_count` is not the right variable to check against here. You wanna do `HOTENDS`